core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleWriter.scala

```diff
@@ -88,7 +88,12 @@ private[spark] class SortShuffleWriter[K, V, C](
     } finally {
       // Clean up our sorter, which may have its own intermediate files
       if (sorter != null) {
+        val startTime = System.nanoTime()
         sorter.stop()
+        context.taskMetrics().shuffleWriteMetrics.getOrElse({
+          metrics : ShuffleWriteMetrics =>
+            metrics.incShuffleWriteTime(System.nanoTime()-startTime)
+        },Nil)
```

Comment:
OK. As a matter of style I think it would be better to...

```
context.taskMetrics.shuffleWriteMetrics.foreach(
          _.incShuffleWriteTime(System.nanoTime - startTime))
```

Which is what `ExternalSorter` does. This looks like the correct bit to time.